transformations/src/main/java/jp/wasabeef/glide/transformations/NinePatchMaskTransformation.java

```diff
@@ -0,0 +1,96 @@
+package jp.wasabeef.glide.transformations;
+
+/**
+ * Copyright (C) 2015 Wasabeef
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.PorterDuff;
+import android.graphics.PorterDuffXfermode;
+import android.graphics.Rect;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.load.Transformation;
+import com.bumptech.glide.load.engine.Resource;
+import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool;
+import com.bumptech.glide.load.resource.bitmap.BitmapResource;
+import com.bumptech.glide.request.target.Target;
+
+public class NinePatchMaskTransformation implements Transformation<Bitmap> {
+
+    private Context mContext;
+    private BitmapPool mBitmapPool;
+    private int mMaskId;
+    private int mWidth;
+    private int mHeight;
+
+    public NinePatchMaskTransformation(Context context, int maskId, int width, int height) {
+        mBitmapPool = Glide.get(context).getBitmapPool();
+        mContext = context;
+        mMaskId = maskId;
+        mWidth = width;
+        mHeight = height;
+    }
+
+    @Override
+    public Resource<Bitmap> transform(Resource<Bitmap> resource, int outWidth, int outHeight) {
+        Bitmap source = resource.get();
+
+        int width = outWidth == Target.SIZE_ORIGINAL ? source.getWidth() : outWidth;
```

Comment:
Because when Glide..override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL), the `mask`'s size may not equal the `outWidth/outHeight`.
if the values not equal, the transform result may not the best one, the image's edges may output sawtooth.

Sorry for my poor English.